python/libsharp/libsharp.pyx

```diff
@@ -4,6 +4,15 @@ __all__ = ['legendre_transform', 'legendre_roots', 'sht', 'synthesis', 'adjoint_
            'analysis', 'adjoint_analysis', 'healpix_grid', 'triangular_order', 'rectangular_order',
            'packed_real_order']
 
+# For compatibility with mpi4py <= 1.3.1
+# For newer versions, one can use the MPI._adressof function
+ctypedef object Comm
+try:
+    from mpi4py.MPI cimport Comm
```

Comment:
Reason I used addressof in the first place was to avoid this, because the cimport always triggers an (unavoidable) import of `mpi4py`, which you really really really do not want if no `comm` is passed in. (And I want this wrapper to work well in non-MPI mode too -- in fact as written this won't even build on machines without mpi4py, which is a blocker for accepting this).

So if mpi4py <= 1.3.1 is really really important then this works, but then we need we need to
- [ ] move this to another `.pyx` module in the same package and conditionally import it only when `comm is not None`, so that mpi4py isn't always imported
- [ ] only build that pyx module if MPI is enabled (logic in build system)
- [ ] I think we need to call mpi4py in the build system to get the proper include paths and pass them to Cython..
- [ ] ...and if we do the above, we might as well check mpi4py version and don't do this for mpi4py >= 1.3.1

So in my book this wasn't worth the effort but if you do the above I'm happy to include it :)